pkg/apis/autoscaling/v1alpha1/metric_validation.go

```diff
@@ -26,7 +26,7 @@ import (
 
 // Validate validates the entire Metric.
 func (m *Metric) Validate(ctx context.Context) *apis.FieldError {
-	errs := serving.ValidateObjectMetadata(m.GetObjectMeta()).ViaField("metadata")
+	errs := serving.ValidateObjectMetadata(ctx, m.GetObjectMeta()).ViaField("metadata")
 	return errs.Also(m.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
```

Comment:
`nil` FieldError can do `Also`, so just
```suggestion
	return serving.ValidateObjectMetadata(ctx, m.GetObjectMeta()).ViaField("metadata")
	  .Also(...)
```